### PR TITLE
updated to new URL & adapted to new content format

### PR DESCRIPTION
**Pull request by richardpenner**
At a glance, this PR attempts to restore functionality to this project by using the new MLB.com probable pitchers URL, and adapting to the new HTML structure at that URL. Additionally, I've updated to a newer `Lab` module, as the old one had memory leaks. Lastly, some additional tests were added. 

Possible issue: I kept the test date the same, but interestingly I am seeing 15 games on that day, whereas the old test was returning 14.

Feedback welcome!

### CODE REVIEW

**Comment by jaw187** [comment]
Since I don't have CI setup, can you just post results of `npm test`

**Comment by richardpenner** [comment]
Passed but I see you have linting setup – I’ll do a cleanup. Node >= 8 OK? If so I’ll add that to package.json too.

**Comment by jaw187** [comment]
ya, Node 8 is fine

**Comment by richardpenner** [comment]
Tests passing, fixed all linter errors. Code coverage is set to 100, and I added a few tests, but the coverage is still not complete – any ideas?

Screen Shot 2019-03-22 at 9 53 53 PM

File: index.js [hunk 1]
```diff
@@ -1,79 +1,99 @@
...
-
-
 const Xray = require('x-ray');
 
+class Probables {
+    static get(dateString, callback) {
 
-const internals = {};
+        let dateComponents = dateString.split('-');
+        if (dateComponents.length !== 3) {
```
Comment (jaw187) [inline index.js:8]: Write a test to cover this code

File: index.js [hunk 2]
```diff
@@ -1,79 +1,99 @@
...
-                away: teams[i],
-                home: teams[j]
-            },
-            pitchers: {
-                away: {
-                    id: pitchers[i],
-                    name: names[i],
-                    throws: throws[i]
+            const matches = pitcherIDInURL.exec(pitcher);
+            if ( matches.length !== 2 ) {
```
Comment (jaw187) [inline index.js]: white space on the `(` and `)`

File: index.js [hunk 3]
```diff
@@ -1,79 +1,99 @@
...
+                timezone: result.timezones[i],
+                teams: {
+                    away: teams[i],
+                    home: teams[j]
                 },
-                home: {
-                    id: pitchers[j],
-                    name: names[j],
-                    throws: throws[j]
+                pitchers: {
```
Comment (jaw187) [inline index.js:78]: This technically makes this a breaking change, but it makes sense

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "mlbprobablepitchers",
-  "version": "1.0.0",
+  "version": "1.1.0",
```
Comment (jaw187) [inline package.json]: we'll version this after the code gets merged

File: index.js (hunk 1 above)
Comment (jaw187) [reply index.js:8]: Nevermind, tests look good.  Not sure what's up with the coverage report.

File: test/index.js [hunk 5]
```diff
@@ -1,29 +1,71 @@
...
-const describe = lab.describe;
-const it = lab.it;
-const expect = Code.expect;
+    const day = '2011/07/23';
+    it(`Gets probables using date syntax ${day}`, (done) => {
 
+        Probables.get(day, (err, dailyMatchups) => {
 
-describe('MLB Probable Pitchers', () => {
+            try {
```
Comment (jaw187) [inline test/index.js]: Is this a new pattern for Lab v15?

**Review by jaw187 — CHANGES_REQUESTED** [review]
Only required change is the white spaces.  But I'm worried that wrapping the test assertions in try/catch is not needed.

File: index.js (hunk 1 above)
Comment (richardpenner) [reply index.js:8]: Yeah, I tried to add coverage, and it seems like I did? But still showing as not covered.

File: index.js (hunk 3 above)
Comment (richardpenner) [reply index.js:78]: sorry – which part breaks it? I'd intended to keep it backward compatible

**Comment by jaw187** [comment]
remove all of the try catches and copy/paste the test results, sorry

**Comment by richardpenner** [comment]
> mlbprobablepitchers@1.0.0 test /Users/richardpenner/PROJECTS/mlbprobablepitchers
> lab -m 9999 -v -L -t 100

MLB Probable Pitchers
  ✔ 1) Gets probables using date syntax 2011/07/23 (46 ms)
  ✔ 2) Gets probables using date syntax 2011-07-23 (2 ms)
  ✔ 3) Fails with invalid date syntax 2011-07 (1 ms)


3 tests complete
Test duration: 61 ms
No global variable leaks detected
Coverage: 68.67% (26/83)
index.js missing coverage on line(s): 8, 10, 11, 34, 35, 36, 37, 38, 44, 46, 47, 52, 54, 55, 57, 58, 59, 61, 63, 64, 65, 67, 68, 69, 92, 95
Code coverage below threshold: 68.67 < 100
Linting results: No issues

**Comment by richardpenner** [comment]
Pushed an update removing the remaining try/catch blocks – I missed those earlier. Note that I added them because I was seeing some REALLY strange behaviour from Xray – when the test was failing, it invoked the Xray callback a second time. I'm not sure I was using the right node version, and I was also invoking the test directly rather than through `npm test`. Anyhow, I think it's working fine now, though the code coverage report is still inaccurate.

File: index.js (hunk 3 above)
Comment (richardpenner) [reply index.js:78]: To be clear, I don't think the JSON format has changed at all – please let me know if it has.

Comment (jaw187) [reply index.js:78]: The object originally only had `home` and `away` properties that had the pitcher details.  Now those two objects are moved into a `pitchers` property.  I don't hate the change, it makes sense

**Review by jaw187 — APPROVED** [review]

**Pull request merged**